v2/spanner-common/src/main/java/com/google/cloud/teleport/v2/spanner/migrations/avro/AvroToValueMapper.java

```diff
@@ -281,6 +282,12 @@ static ByteArray avroFieldToByteArray(Object recordValue, Schema fieldSchema) {
         }
         return ByteArray.copyFrom(Hex.decodeHex(s));
       }
+      if (fieldSchema.getType().equals(Schema.Type.BYTES)) {
+        // For a bytes avro type, expect a java.nio.ByteBuffer.
+        if (recordValue instanceof ByteBuffer) {
+          return ByteArray.copyFrom(((ByteBuffer) recordValue).array());
+        }
+      }
       return ByteArray.copyFrom((byte[]) recordValue);
```

Comment:
This was added as a catchall for byte types but currently for mysql none of them are using this flow AFAIK. We should make the default catchall as ByteBuffer if that is the avro standard.

Reply:
Done, I was worried that some path was using it so I did not default to it.